File: frontend/src/services/EmployeeService.js
import axios from 'axios'

const apiUrl = 'http://localhost/api/v1/employees';

class EmployeeService {

    getAllEmployees() {
        return axios.get(apiUrl);
    }

    createEmployee(employee) {
        return axios.post(apiUrl,apiUrl);
    }

    getEmployeeById(employeeId) {
        return axios.get(apiUrl + '/' + employeeId);
    }

    updateEmployee(employeeId,employee) {
        return axios.put(apiUrl + '/' + employeeId, employee);
    }

    deleteEmployee(employeeId) {
        return axios.delete(apiUrl + '/' + employeeId);
    }
}
export default new EmployeeService();

File: backend/src/main/java/net/javaguides/springboot/service/EmployeeService.java
package net.javaguides.springboot.service;

import net.javaguides.springboot.model.Employee;

import java.util.List;

public interface EmployeeService {

    Employee createEmployee(Employee employee);

    Employee getEmployeeById(Long employeeId);

    List<Employee> getAllEmployees();

    Employee updateEmployee(Long employeeId, Employee updatedEmployee);

    void deleteEmployee(Long employeeId);
}
